.github/workflows/pr.yml

```diff
@@ -58,18 +57,44 @@ jobs:
         # 64 bit int
         cmake ${SHARED_OPT} -DBUILD_INDEX64=on -B lapack/build64
         cmake --build lapack/build64 ${PARALLEL} --target install
+    - name: Test if the changes affect this domain
+      id: test_domain
+      uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+      with:
+        script: |
+          function matchesPattern(filePaths) {
+            console.log('files:', filePaths)
+            // These directories contain domain specific code
+            const dirs = '(tests/unit_tests|examples|src|include/oneapi/mkl)'
+            const domains = '(blas|lapack|rng|dft)'
+            const domain = '${{ matrix.domain }}'
+            // matches changes to the domain of interest or non domain-specific code
+            const re = new RegExp(`^(${dirs}/${domain}|(?!${dirs}/${domains}))`);
```

Comment:
Would it be a match if the changes are in `docs`, `scripts`, or `third-party-programs` directories?

Reply:
Yes. It would match. Now that the  logic is javascript and a regular expression, you can make the filter as selective as you want. It is straightforward to exclude files that are certain not to affect the build like `*.md`. If the team wants to fine tune the selection logic, I would first move the javascript to a separate file so it could more easily be tested outside the CI system. 

I would rather do it in a separate PR. I want to propose adding some static checkers and I want to get that started with a PR because it might take a while to settle on something that the team is happy with.

Reply:
Thank you!